### ISSUE TITLE
BungeePackFix 1.1.3 When using Minecraft 1.20.2, switching servers does not skip resource pack loading, it always reloads

### ISSUE THREAD

**Issue by tygh89071388** [LoneDev6/BungeePackFix#44]
When using Minecraft 1.20.2, switching servers does not skip resource pack loading, it always reloads

This server is running Paper version git-Paper-169 (MC: 1.20.1) (Implementing API version 1.20.1-R0.1-SNAPSHOT) (Git: b4e3b3d)
This server is running Waterfall version git:Waterfall-Bootstrap:1.20-R0.2-SNAPSHOT:0e657ed:545 by md_5

**Comment by SrFirox** [comment]
i have the same problem.

**Comment by LoneDev6** [comment]
Check if you set ignore hash `false` in config.yml of BungeePackFix, it might be the reason.
The plugin was tested and working before publishing the update.

**Comment by naikyys** [comment]
Hey, its not misconfigured. Have the same issue, only 1.20.2 is still redownloading the pack, but lower versions not. Even if I set ignore hash to false in config.yml of BungeePackFix. Please fix it :/

**Comment by LoneDev6** [comment]
Can you confirm that 1.20.2 requires ignore hash to be set to false in order to avid redownloading the pack each time?

**Comment by naikyys** [comment]
> Can you confirm that 1.20.2 requires ignore hash to be set to false in order to avid redownloading the pack each time?

No. You have to just fix that 1.20.2 is still redownloading and versions below 1.20.2 are fine.

**Comment by SuprexDE** [comment]
I have the same issue. It makes no difference whether you set ignore hash to "true" or "false".

**Comment by SuprexDE** [comment]
i have the same issue. When will the fix come?

**Comment by naikyys** [comment]
Is it please possible to fix it please?... As you can see, many people having the same issue.

**Comment by nhanledev** [comment]
I created a pull request at https://github.com/LoneDev6/BungeePackFix/pull/48 and that should fix this issue.

**Comment by LoneDev6** [comment]
It's related to a Minecraft bug which we can't fix for now.
https://bugs.mojang.com/browse/MC-265673

**Comment by Domm1** [comment]
It got suggested to feedback.minecraft since they deem this as feature request, here is the link to vote for it. 

https://feedback.minecraft.net/hc/en-us/community/posts/20698154329869-Don-t-reset-the-resource-pack-when-entering-configuration-phase

**Comment by jhqwqmc** [comment]
https://www.minecraft.net/en-us/article/minecraft-1-20-3-pre-release-1

**Comment by LoneDev6** [comment]
Did they fix the issue? Did they revert the logic or changed it again?

**Comment by jhqwqmc** [comment]
Maybe changed a little logic, also restored some logic
![image](https://github.com/LoneDev6/BungeePackFix/assets/113768629/3a8ed6ff-61b2-4470-9089-6fadd0d12d92)

**Comment by derMiepz** [comment]
Seems like Mojang already fixed the issue with reloading texture packs, so fixing it shouldn't be that big of a deal.

**Comment by nhanledev** [comment]
They (Mojang) change the resource pack logic completely to other packet from 1.20.3, so Lone has to rewrite some additional codes to fix that.

**Comment by el211** [comment]
> 1.20.2 is a bugged version and it's not planned to be added as compatibility because people should stop using it. Use 1.20.3 instead. 1.20.3 support is added in the new Bungeepackfix update. 1.20.2 clients won't be handled by Bungeepackfix anymore so the plugins can send packs anytime, so clients should correctly see the pack sent each time they change server.
> 
> Let me know if it works on 1.20.3+ Download: https://github.com/LoneDev6/BungeePackFix/releases/tag/1.1.5-beta-1

i got same issue... cant use 1.20.3 this will break all my custom maps....

**Comment by ttt0** [comment]
> 1.20.2 is a bugged version and it's not planned to be added as compatibility because people should stop using it. Use 1.20.3 instead. 1.20.3 support is added in the new Bungeepackfix update. 1.20.2 clients won't be handled by Bungeepackfix anymore so the plugins can send packs anytime, so clients should correctly see the pack sent each time they change server.
> 
> Let me know if it works on 1.20.3+ Download: https://github.com/LoneDev6/BungeePackFix/releases/tag/1.1.5-beta-1


It does not work properly in version 1.20.4. I use Flamecord and in most of the things I do, it gives the error "disconnected with: DecoderException: io.netty.util.IllegalReferenceCountException: refCnt: ​​0, decrement: 1 @ io.netty.handler.codec.MessageToMessageDecoder:98" and throws it out of the game. It gets fixed when I delete bungeepackfix.

**Comment by victordumoulin** [comment]
Hello; i use 1.20.4 but the plugin on the bungee doesn't seem to create an error; but the texture pack reloads between servers..
Does anyone have a solution?? even a crazy one
![Sans titre](https://github.com/LoneDev6/BungeePackFix/assets/60874199/e73dbf43-7f03-458d-9d04-5a7ee512cb11)

**Comment by TerryLVM** [comment]
1.20.3 and 1.20.4 are working fine, but previous versions such as 1.19.4 or 1.20.1 stopped working.
It seems that you removed support for those versions because the resourcepack is sent each time they change server.

**Comment by ttc111** [comment]
1.20.4 will repeatedly load texture packs.

**Comment by TerryLVM** [comment]
These are my settings:

messages:
  minimessage_support: false
  ignored_pack:
    enabled: true
    message: "&7Texturepack cargado con éxito"
equal_pack_attributes:
  hash: true
  forced: true
  prompt_message: true
  modern_uuid: true
ignore_hash_in_url: true
main_spigot_server_name: lobby1
ignored_servers:
  enabled: false
  invert_check: false
  list:
    - MULTIFFA-1
    - DUELS-1
    - DUELS-2
log:
  ignored_respack: false
  sent_respack: false
  debug: false
cancel_modern_resourcepack_remove_packet: true
